File: rank_test_problems_b_0001/rank_test_problems_b_0001__4.py
# 実行時間 10ms
i = 0
s = input()

while i <= len(s) - 5:
    t = s[i : i + 5]

    if t == "LLLRB":
        i += 5
        print("rolling")
    elif t == "DDRRA":
        i += 5
        print("upper")
    elif t == "AAAAA":
        i += 5
        print("rush")
    else:
        i += 1

File: rank_test_problems_b_0001/rank_test_problems_b_0001__4.cs
// 実行時間 20ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        string s = ReadLine();
        var c = System.StringComparison.Ordinal;

        for( int i = 0; i <= s.Length - 5; ) {
            string t = s.Substring( i, 5 );

            if( string.Compare( t, "LLLRB", c ) == 0 ) {
                i += 5;
                WriteLine( "rolling" );
            } else if( string.Compare( t, "DDRRA", c ) == 0 ) {
                i += 5;
                WriteLine( "upper" );
            } else if( string.Compare( t, "AAAAA", c ) == 0 ) {
                i += 5;
                WriteLine( "rush" );
            } else {
                i++;
            }
        }
        return;
    }
}
